lib/config/fileconf.go

```diff
@@ -1084,9 +1085,28 @@ func (a *AuthenticationConfig) Parse() (types.AuthPreference, error) {
 		}
 	}
 
+	if a.SecondFactor != "" {
+		secondFactors := types.SecondFactorsFromLegacySecondFactor(a.SecondFactor)
+		var secondFactorStrings []string
+		for _, sf := range secondFactors {
+			secondFactorStrings = append(secondFactorStrings, sf.ToString())
+		}
+
+		log.Warnf(``+
+			`The "second_factor" setting is marked for removal in favor of second_factors. `+
+			`Please update your configuration to use second_factors. e.g. "second_factors: %v".`, secondFactorStrings)
+
+		if a.SecondFactors != nil {
+			log.Warnf(`` +
+				`second_factor and second_factors are both set. second_factors will take precedence. ` +
+				`second_factor should be unset to remove this warning.`)
+		}
+	}
+
 	return types.NewAuthPreferenceFromConfigFile(types.AuthPreferenceSpecV2{
 		Type:                    a.Type,
 		SecondFactor:            a.SecondFactor,
+		SecondFactors:           a.SecondFactors,
```

Comment:
Similar to other comments, won't this cause changes to CAP?

Reply:
I'm not sure what you mean

Reply:
I mean that we are writing both second_factor and second_factors to the "cluster_auth_preference". This means that, on a read, we'll return a resource that has both fields set and thus is invalid by Teleport's own criteria.

This is all to say I would rather keep the cap as the user wrote it and calculate whatever we need in the getters, on the fly.

Reply:
This _is_ taking the cap as the user wrote it. But I will remove the warning log and update `CheckAndSetDefaults` to return `trace.BadParameter("must set either SecondFactor or SecondFactors, not both")` sot that this would return an error instead.

Reply:
Right, sorry for the noise.